pyvista/core/composite.py

```diff
@@ -232,7 +234,8 @@ def center(self) -> NumpyArray[float]:
 
         """
         # (typing.cast necessary to make mypy happy with np.reshape())
-        return np.reshape(cast(List[float], self.bounds), (3, 2)).mean(axis=1)
+        center_array = np.reshape(cast(List[float], self.bounds), (3, 2)).mean(axis=1)
```

Comment:
The `mypy` config warns if the cast is not needed, so this must still be required.
https://github.com/pyvista/pyvista/blob/4cdd4eacbcf6d63b1dc5540b16724779805447e2/.pre-commit-config.yaml#L59

Reply:
Since `self.bounds` is a tuple, we are doing tuple->array for `reshape`, and then array->list->tuple for the output. So, we can instead forego the use of numpy and do what `Prop3DMixin` does:
https://github.com/pyvista/pyvista/blob/4cdd4eacbcf6d63b1dc5540b16724779805447e2/pyvista/plotting/prop3d.py#L578-L585

Reply:
I changed the line to `center_array = np.reshape(self.bounds, (3, 2)).mean(axis=1)` and mypy run through pre-commit `pre-commit run mypy --all-files` is passing for me.

Reply:
I'm beginning to trust mypy less and less through pre-commit.  I tried changing the other line to `cast(Tuple[int, int, int], tuple(center_array.tolist()))` and it is also passing for me.  Weird...

Reply:
Very odd indeed. Sometimes `mypy` won't type check if the signature is not typed, i.e. no annotations for any args or return type. But here the return type _is_ annotated as `tuple[float, float, float]`. And just to be sure it's not just the args that must be typed, I also tried annotating `self` with `def center(self: MultiBlock)`, but this doesn't raise mypy errors either. In fact, you can directly return a numpy array or a list without any casting and mypy still won't complain. Annotating the array variable inside the function with `center_array: NumpyArray[float]` also still doesn't help, and no warnings about redundant casts and/or the wrong return type are generated.

If I try this:
``` python
out = tuple(np.reshape(self.bounds, (3, 2)).mean(axis=1).tolist())
reveal_type(out)
return out
```
and run `mypy pyvista | grep Revealed`, I get ``` Revealed type is "builtins.tuple[Any, ...]" ```, which is what I would have expected. But still, mypy clearly knows the type, and yet isn't complaining that this doesn't match `tuple[float, float, float]`. Adding `cast` (and running `mypy pyvista | grep composite`) also doesn't generate warnings about redundant casts. So, this issue may not be specific to pre-commit.

Reply:
If it passes through the type checking without the cast, then we should get rid of it IMO.

Reply:
> If it passes through the type checking without the cast, then we should get rid of it IMO.

Removed in https://github.com/pyvista/pyvista/pull/6469/commits/09680766fc2187db56149d9049e37af7e344c569